session.go

```diff
@@ -1266,27 +1273,63 @@ func (s *session) dropEncryptionLevel(encLevel protocol.EncryptionLevel) {
 	}
 }
 
+// is called for the client, when restoring transport parameters saved for 0-RTT
+func (s *session) restoreTransportParameters(params *wire.TransportParameters) {
+	if s.logger.Debug() {
+		s.logger.Debugf("Restoring Transport Parameters: %s", params)
+	}
+
+	s.peerParams = params
+	s.connIDGenerator.SetMaxActiveConnIDs(params.ActiveConnectionIDLimit)
+	s.connFlowController.UpdateSendWindow(params.InitialMaxData)
+	if err := s.streamsMap.UpdateLimits(params); err != nil {
+		s.closeLocal(err)
+		return
+	}
+}
+
 func (s *session) processTransportParameters(params *wire.TransportParameters) {
+	if err := s.processTransportParametersImpl(params); err != nil {
+		s.closeLocal(err)
+	}
+}
+
+func (s *session) processTransportParametersImpl(params *wire.TransportParameters) error {
 	if s.logger.Debug() {
 		s.logger.Debugf("Processed Transport Parameters: %s", params)
 	}
 	if s.qlogger != nil {
 		s.qlogger.ReceivedTransportParameters(params)
 	}
 
-	// check the Retry token
-	if s.perspective == protocol.PerspectiveClient && !params.OriginalConnectionID.Equal(s.origDestConnID) {
-		s.closeLocal(qerr.NewError(qerr.TransportParameterError, fmt.Sprintf("expected original_connection_id to equal %s, is %s", s.origDestConnID, params.OriginalConnectionID)))
-		return
+	// check the initial_source_connection_id
+	if !params.InitialSourceConnectionID.Equal(s.handshakeDestConnID) {
+		return qerr.NewError(qerr.TransportParameterError, fmt.Sprintf("expected initial_source_connection_id to equal %s, is %s", s.handshakeDestConnID, params.InitialSourceConnectionID))
```

Comment:
Could exposing the expected id leak info to an attacker?

Reply:
Not really. The connection ID has already been used on the Initial packet, so it's known to a MITM.